Sources/SwiftBSON/BSONBinary.swift

```diff
@@ -89,13 +89,21 @@ public struct BSONBinary: Equatable, Hashable {
         self = try BSONBinary(buffer: buffer, subtype: subtype)
     }
 
-    internal init(bytes: [UInt8], subtype: Subtype) throws {
+    /// Initializes a `BSONBinary` instance from a `[UInt8]` array and a `Subtype` subtype.
+    /// This will always create a copy of the array contents.
+    /// - Throws:
+    ///   - `BSONError.InvalidArgumentError` if the provided array is incompatible with the specified subtype.
+    public init(bytes: [UInt8], subtype: Subtype) throws {
         var buffer = BSON_ALLOCATOR.buffer(capacity: bytes.count)
         buffer.writeBytes(bytes)
         self = try BSONBinary(buffer: buffer, subtype: subtype)
     }
 
-    internal init(buffer: ByteBuffer, subtype: Subtype) throws {
+    /// Initializes a `BSONBinary` instance from a `ByteBuffer` and a `Subtype` subtype.
+    /// This will always create a copy of the buffer contents.
```

Comment:
I'm not sure if this line is true in this case, because I believe `ByteBuffer` implements copy-on-write semantics (see various uses of `isKnownUniquelyReferenced` here: https://github.com/apple/swift-nio/blob/main/Sources/NIOCore/ByteBuffer-core.swift) 
so maybe we should just remove this line?

Reply:
ah great catch, i had just copied that sentence over blindly